tests/lib/tools/log-parser

```diff
@@ -411,7 +436,7 @@ class LogReader:
             extra = None
         elif info_type == DEBUG_TYPE:
             verb = parts[3]
-            task = parts[4]
+            task = parts[5]
```

Comment:
I wonder if this and `info_type == ERROR_TYPE` is maybe missing a check that `len(parts) >= 6` (or 5 for the ERROR_TYPE).

Reply:
In this case the task is the 5th part of the string:
 2021-11-04 09:59:53 Debug output for google:ubuntu-core-18-64:tests/main/snap-connections (nov040945-011581) : 

for Error is different 
2021-11-04 09:59:52 Error executing google:ubuntu-core-18-64:tests/main/snap-connections (nov040945-011581) : 

Now I added examples in all the functions to easier to understand.